src/pages/index.js

```diff
@@ -20,80 +20,222 @@ import {
 
-function handleImageClick(name, link) {
-  popupWithImage.open({ name, link });
+  likeRequest
+    .then((updatedCard) => {
+      card.setLikes(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error("Error updating like status:", err);
+      card.setLikes(isLiked);
+    });
 }
 
+// Function to create a new card
 function createCard(cardData) {
   const card = new Card(
     cardData,
     "#card-template",
-
-    handleImageClick
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike,
+    currentUserId
   );
-  return card.getView();
+
+  const cardElement = card.getView();
+  cardElement.setAttribute("data-id", cardData._id);
+
+  return cardElement;
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".cards__list"
+// Function to handle image clicks and open the preview
+function handleImageClick(name, link) {
+  popupWithImage.open({ name, link });
+}
+
+// Popup handling
+const profilePopup = new PopupWithForm("#profile-edit-modal", (formData) => {
+  const updatedName = formData["profile-title-input"];
+  const updatedAbout = formData["profile-subtitle-input"];
+
+  profilePopup.setLoadingText(true); // Show "Saving..."
+
+  api
+    .updateUserProfile(updatedName, updatedAbout)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        job: updatedUserData.about,
+      });
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profilePopup.setLoadingText(false, "Save");
+    });
+});
+
+const avatarPopup = new PopupWithForm(
+  "#modal-change-profile-picture",
+  (formData) => {
+    console.log("formData:", formData);
+
+    const newAvatarLink = formData["profile-picture-link"];
+
+    avatarPopup.setLoadingText(true);
+
+    api
+      .updateAvatar(newAvatarLink)
+      .then((updatedUser) => {
+        userInfo.setUserAvatar(updatedUser.avatar);
+        avatarPopup.close();
+      })
+      .catch((err) => {
+        console.error("Error updating avatar:", err);
+      })
+      .finally(() => {
+        avatarPopup.setLoadingText(false, "Save");
+      });
+  }
 );
-cardSection.renderItems();
+avatarPopup.setEventListeners();
 
+avatarEditButton.addEventListener("click", () => {
+  avatarFormValidator.disableSubmitButton();
+  avatarPopup.open();
+});
+
+const cardPopup = new PopupWithForm("#card-add-modal", (formData) => {
+  const cardName = formData["card-title-input"];
+  const cardLink = formData["card-link-input"];
+
+  const cardData = {
+    name: cardName,
+    link: cardLink,
+  };
+
+  cardPopup.setLoadingText(true);
+
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      renderCard(newCard);
+      cardPopup.close();
```

Comment:
* You should **clear the inputs** after a successful adding of a new card, to let the user add the 2nd one again without having to remove the old data manually.  You need to **reset** the form only  in a `then` block 


```js
      cardPopup.close();
      cardAddForm.reset()

```